### PR TITLE
Fix bug in utf8 editing & set cursor

### PR DESCRIPTION
**Pull request by aynakeya**
Fix problem with delete, insert, cursor move when editing utf8 runes.

Related: https://github.com/jroimartin/gocui/pull/74 https://github.com/jesseduffield/gocui/pull/10

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,10 +1,9 @@
-module github.com/awesome-gocui/gocui
+module github.com/aynakeya/gocui
```
Comment (mjarkk) [inline go.mod:1]: Can you restore this change?

File: edit.go [hunk 2]
```diff
@@ -183,7 +205,7 @@ func (v *View) moveCursor(dx, dy int) {
 	if newX > len(line) {
 		if dy == 0 && newY+1 < len(v.lines) {
 			newY++
-			// line = v.lines[newY] // Uncomment if adding code that uses line
+			//line = v.lines[newY] // Uncomment if adding code that uses line
```
Comment (mjarkk) [inline edit.go:208]: Can you restore this change?

File: edit.go [hunk 3]
```diff
@@ -156,8 +156,30 @@ func (v *View) EditNewLine() {
...
+			if newX <= col && dx >= 0 {
+				dx = runewidth.RuneWidth(line[index].chr)
+				break
+			}
+			if newX < col && dx < 0 {
+				dx = -runewidth.RuneWidth(line[index].chr)
+				break
+			}
+		}
+	}
```
Comment (mjarkk) [inline edit.go:181]: Can you add a comment to what this is doing?

File: edit.go [hunk 4]
```diff
@@ -261,6 +283,14 @@ func (v *View) writeRune(x, y int, ch rune) error {
 		chr:     ch,
 	}
 
+	for i := 1; i < w; i++ {
+		v.lines[y][x+i] = cell{
+			fgColor: v.FgColor,
+			bgColor: v.BgColor,
+			chr:     '\x00',
+		}
+	}
```
Comment (mjarkk) [inline edit.go:293]: Can you add a comment to what this is doing?

File: go.mod [hunk 5]
```diff
@@ -1,10 +1,9 @@
...
 
 go 1.13
 
 require (
 	github.com/gdamore/tcell/v2 v2.0.0
 	github.com/mattn/go-runewidth v0.0.9
-	golang.org/x/sys v0.0.0-20200930185726-fdedc70b468f // indirect
```
Comment (mjarkk) [inline go.mod:8]: I don't think we need this change

File: edit.go [hunk 6]
```diff
@@ -6,7 +6,6 @@ package gocui
 
 import (
 	"errors"
-
```
Comment (mjarkk) [inline edit.go:9]: Can you restore this change

File: edit.go [hunk 7]
```diff
@@ -156,8 +156,30 @@ func (v *View) EditNewLine() {
...
+				break
+			}
+			if newX < col && dx < 0 {
+				dx = -runewidth.RuneWidth(line[index].chr)
+				break
+			}
+		}
+	}
 	v.moveCursor(dx, dy)
-	v.gui.userEvents <- userEvent{func(g *Gui) error { return nil }}
```
Comment (mjarkk) [inline edit.go:160]: Why did you remove this line?

**Review by mjarkk — CHANGES_REQUESTED** [review]

**Comment by aynakeya** [comment]
sry i forget to create a new branch. gonna create a new pull request again.
